src/types/utils.ts

```diff
@@ -116,6 +116,15 @@ type TrimRight<T, Chars extends string = ' '> = T extends `${infer R}${Chars}`
   ? TrimRight<R>
   : T
 
+/**
+ * @description Creates a type with required keys K from T.
+ *
+ * @example
+ * RequiredBy<{ a?: string, b?: number, c: number }, 'a' | 'c'>
+ * => { a: string, b?: number, c: number }
```

Comment:
Recently trying to use `// ^?` for `@example`s

```suggestion
 * type Result = RequiredBy<{ a?: string, b?: number, c: number }, 'a' | 'c'>
 * //   ^? { a: string, b?: number, c: number }
```